### ISSUE TITLE
GameConqueror: "Manually add an Entry" menu breaks when closed with "X"

### ISSUE THREAD

**Issue by 12345ieee** [scanmem/scanmem#160]
Already reported in https://github.com/scanmem/scanmem/issues/149#issuecomment-171916851
Steps to reproduce:

* Open the "Manually add an entry" menu
* Close it with the "X"
* Reopen it
* Bug triggered
* Closing it with the "Close" button gives no problem.

**Comment by sriemer** [comment]
GTK error messages on Ubuntu 14.04 trusty:
````
(GameConqueror.py:9016): Gtk-CRITICAL **: gtk_container_foreach: assertion 'GTK_IS_CONTAINER (container)' failed

(GameConqueror.py:9016): Gtk-CRITICAL **: gtk_container_set_border_width: assertion 'GTK_IS_CONTAINER (container)' failed

(GameConqueror.py:9016): Gtk-CRITICAL **: _gtk_box_get_spacing_set: assertion 'GTK_IS_BOX (box)' failed

(GameConqueror.py:9016): Gtk-CRITICAL **: gtk_box_set_spacing: assertion 'GTK_IS_BOX (box)' failed

(GameConqueror.py:9016): Gtk-CRITICAL **: _gtk_box_set_spacing_set: assertion 'GTK_IS_BOX (box)' failed

````

**Comment by sriemer** [comment]
The line number is invalid. But this is the AddCheatDialog (self.addcheat_*) initialized in [line 263 of GameConqueror.py](https://github.com/scanmem/scanmem/blob/d20e3e495477b9a1aa38682a21bbbb59c5d967a8/gui/GameConqueror.py#L263).

**Comment by sriemer** [comment]
Okay, I've found the bug: The `self.addcheat_dialog` is only initialized once. Clicking at the button calls `self.addcheat_dialog.show()` and clicking at "confirm" or "close" in the dialog calls `self.addcheat_dialog.hide()`. But clicking at the X results in deletion of the dialog it seems.

**Comment by 12345ieee** [comment]
Have a look: http://faq.pygtk.org/index.py?req=show&file=faq10.021.htp

**Comment by sriemer** [comment]
Nice! It's working! Thanks!

````
$ git diff
diff --git a/gui/GameConqueror.py b/gui/GameConqueror.py
index f9205b0e6508..e3febe0ac974 100644
--- a/gui/GameConqueror.py
+++ b/gui/GameConqueror.py
@@ -266,6 +266,7 @@ class GameConqueror():
         self.addcheat_type_combobox = self.builder.get_object('Type_ComboBox')
         misc.build_combobox(self.addcheat_type_combobox, LOCK_VALUE_TYPES)
         misc.combobox_set_active_item(self.addcheat_type_combobox, SETTINGS['lock_data_type'])
+        self.addcheat_dialog.connect('delete-event', lambda acd, e: acd.hide() or True)
         
         
         
````
I'll do the commit.

**Comment by sriemer** [comment]
The fix is committed. Thanks!
Please test latest code from the `master` branch as well and confirm that this is fixed.

**Comment by sriemer** [comment]
I've released scanmem 0.15.7-0sp2 to my PPA with the new patches from today applied via quilt.

**Comment by 12345ieee** [comment]
Thank you, I confirm the issue is fixed (PPA version installed on Debian sid).
